project/src/components/favorites-screen/favorites-screen.tsx

```diff
@@ -1,28 +1,40 @@
-function FavoritesScreen(): JSX.Element {
+import { nanoid } from '@reduxjs/toolkit';
+import { Offer } from '../../types/offer';
+import FavoritesOffersLst from '../favorites-offers-list/favorites-offers-list';
+import { Link } from 'react-router-dom';
+import { AppRoute } from '../../const';
+
+type FavoritesScreenProps = {
+  offers: Offer[],
+}
+
+function FavoritesScreen({ offers }: FavoritesScreenProps): JSX.Element {
+  const favoriteLocations: Set<string> = new Set();
+  offers.slice().filter((offer) => offer.isFavorite).forEach((offer) => favoriteLocations.add(offer.cityName));
```

Comment:
```suggestion
  offers.filter((offer) => offer.isFavorite).forEach((offer) => favoriteLocations.add(offer.cityName));
```
ничего не меняй по ссылке
сделай условие в forEach просто